Nodejs/Tests/AnalysisTests/AnalysisFile.cs

```diff
@@ -131,6 +131,21 @@ static class Analysis {
             List<AnalysisFile> files = new List<AnalysisFile>();
             foreach (var file in Directory.GetFiles(directory, "*", SearchOption.AllDirectories)) {
                 if (String.Equals(Path.GetExtension(file), ".js", StringComparison.OrdinalIgnoreCase)) {
+                    var relativeFile = file.Substring(directory.Length);
+                    int nestedModulesCount = 0;
+                    int index = relativeFile.IndexOf("node_modules");
+                    while (index != -1)
+                    {
+                        nestedModulesCount++;
+                        relativeFile = relativeFile.Substring(index + 1);
```

Comment:
instead of creating a whole new string, you can change the search position in IndexOf